src/library/scanner/recursivescandirectorytask.cpp

```diff
@@ -79,14 +79,11 @@ void RecursiveScanDirectoryTask::run() {
     if (prevHash != newHash) {
         // Rescan that mofo! If importing fails then the scan was cancelled so
         // we return immediately.
-        if (!filesToImport.isEmpty()) {
```

Comment:
I thought about that. I lke that in my solution only one class had the responsibility to emit a signal. I think that makes the code easier to understand. .(the whole task scheduling here is already hard to follow). 

I haven't noticed that the no-op task would slow down the scanning for me. So I would like to keep it that way unless it is a huge performance hit for older/slow hardware.